### DISCUSSION TITLE
Is there a github action that can verify `renovate.json` before merging it?

### DISCUSSION THREAD

**Discussion by karfau** [renovatebot/renovate#10630] [Request Help] [answered]
How often have you experienced the following:
You made a change to the renovate config, filed a PR, landed it and right afterwards Renovate (re)opens the issue telling you that your config has an issue.
I want to find a way to prevent this.

In the https://github.com/bettermarks/renovate-config repository I created a script and run it per branch to validate all the presets,
but the execution time is awful (>30s just for installing renovate dependency), and it's to much code to replicate into every repo.

Since I think many people would love to validate their config if it was changed, before landing a PR, some docker based github action would be nice (that doesn't build on every run, but uses the [slim] docker images already provided by renovate) that would handle all the details and adds a potentially failing check.

Has anybody done something like this before and I was just not able to find it?
Is that something that could be provided (and maintained) by renovate, so it keeps working when things change?

Related discussion: #8113
Maybe related: #10394

**Comment by HonkingGoose** [comment]
I found this feature request: https://github.com/renovatebot/renovate/issues/547, quote from top post by the head maintainer of Renovate bot:

> Consider the case when someone has already activated Renovate but wants to make a configuration change. My idea:
> 
> They create an issue "Reconfigure Renovate".
> Renovate bot looks for this each run
> If found, it opens a "Reconfigure Renovate" PR to close that issue.
> PR is initially "empty" but user can edit renovate.json in branch and see results just like the onboarding PR.
> 
> Essentially, it's an interactive way to update renovate configuration and helps them validate the config.

Would having this feature help you out?

**Reply by karfau** [reply]
The idea of opening an issue in the repo where change is needed might be nice, but maybe there could also be other "entry points" that are closer to the action somebody would naturally do: like creating a branch (starting with the configured branch prefix?) that makes a change to renovate.json, or creating a PR mentioning the renovatebot? Or even adding a commit to any branch but the default one that mentions the bot?

Or, following the idea of the "initially empty PR": having a checkbox in the dependency dashboard, the original onboarding PR or any of the created PRs that can be checked to create such a branch/PR.

**Comment by karfau** [comment, marked as answer]
I found that there is a github action, but it doesn't support config validation yet:
https://github.com/renovatebot/github-action/issues/548
